### DISCUSSION TITLE
Render prompt at the bottom of screen

### DISCUSSION THREAD

**Discussion by hanzomaster** [JanDeDobbeleer/oh-my-posh#3872] [Troubleshoot] [answered]
I recently follow this Stack Overflow [post](https://stackoverflow.com/a/23671155/11399551) to render my prompt at the bottom of the screen. However, when I follow the answer and achieve what I want, I can't seem to render my custom oh-my-posh theme. Here what I tried:
```powershell
function prompt {
  # put cursor at the bottom of the buffer
  $rawUI = (Get-Host).UI.RawUI
  $cp = $rawUI.CursorPosition
  $cp.Y = $rawUI.BufferSize.Height - 1
  $rawUI.CursorPosition = $cp

  Set-PoshPrompt
}
```
This one only renders a `PS>` at the bottom of the screen
![Screenshot 2023-05-18 182449](https://github.com/JanDeDobbeleer/oh-my-posh/assets/49803999/4390e35d-af37-432b-bb40-03568e531213)
And this
```powershell
function prompt {
  # put cursor at the bottom of the buffer
    $rawUI = (Get-Host).UI.RawUI
    $cp = $rawUI.CursorPosition
    $cp.Y = $rawUI.BufferSize.Height - 1
    $rawUI.CursorPosition = $cp

  # and the prompt itself
    oh-my-posh --init --shell pwsh --config E:\tmp\my-custom-theme.json | Invoke-Expression
}
```
renders nothing at first and reset to the original prompt after a command
Does anyone know how to render the prompt at the bottom of the screen, or anyway to modify my custome file theme for it to render as I wish it would?

**Comment by JanDeDobbeleer** [comment, marked as answer]
You're not printing the prompt but interpreting the init script that will override the prompt once again.

You want to hijack the prompt function so it will stick at the bottom. Initialize oh-my-posh outside of your custom prompt function, store the value of that function and invoke the original prompt function at the end of your custom function.

VSCode does the same for its shell integrations, you can have a look at that [here](https://github.com/microsoft/vscode/blob/2ca206a703b92e5a186153aa3bc6e789c54201a6/src/vs/workbench/contrib/terminal/browser/media/shellIntegration.ps1#L103).

**Reply by JanDeDobbeleer** [reply]
Something in the lines of this:

```powershell
oh-my-posh --init --shell pwsh --config E:\tmp\my-custom-theme.json | Invoke-Expression
$Global:__OriginalPrompt = $function:Prompt

function prompt {
    # put cursor at the bottom of the buffer
    $rawUI = (Get-Host).UI.RawUI
    $cp = $rawUI.CursorPosition
    $cp.Y = $rawUI.BufferSize.Height - 1
    $rawUI.CursorPosition = $cp

    # and the prompt itself
    $Global:__OriginalPrompt.Invoke()
}

```

Needless to say this voids all warranty. There could be a lot of side effects.

**Reply by hanzomaster** [reply]
Wow that is incredibly useful. Thank you!
